src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -684,36 +684,54 @@ protected void SelectVersion(NuGetVersion latestVersion = null)
                     _versions.IndexOf(SelectedVersion) > _versions.IndexOf(_versions.FirstOrDefault(v => v != null && !v.IsValidVersion))))
             {
                 // The project level is the only one that has an editable combobox and we can only see one project.
-                if (!IsSolution && _nugetProjects.Count() == 1 && FirstDisplayedVersion == null && _nugetProjects.First().ProjectStyle.Equals(ProjectModel.ProjectStyle.PackageReference))
+                if (!IsSolution && _nugetProjects.First().ProjectStyle.Equals(ProjectModel.ProjectStyle.PackageReference))
                 {
-                    // For the Updates and Browse tab we select the latest version, for the installed tab
-                    // select the installed version by default. Otherwise, select the first version in the version list.
-                    IEnumerable<DisplayVersion> possibleVersions = _versions.Where(v => v != null);
-                    if (_filter.Equals(ItemFilter.UpdatesAvailable) || _filter.Equals(ItemFilter.All))
-                    {
-                        SelectedVersion = possibleVersions.FirstOrDefault(v => v.Range.OriginalString.Equals(latestVersion.ToString(), StringComparison.OrdinalIgnoreCase));
-                        UserInput = SelectedVersion.ToString();
-                    }
-                    else
-                    {
-                        SelectedVersion =
-                            possibleVersions.FirstOrDefault(v => StringComparer.OrdinalIgnoreCase.Equals(v.Range?.OriginalString, _searchResultPackage?.AllowedVersions?.OriginalString))
-                            ?? possibleVersions.FirstOrDefault(v => v.IsValidVersion);
-                        UserInput = _searchResultPackage.AllowedVersions?.OriginalString ?? SelectedVersion.ToString();
-                    }
-
-                    FirstDisplayedVersion = SelectedVersion;
+                    SelectVersionPackageReferenceProject(latestVersion);
                 }
                 else
                 {
-                    var possibleVersions = _versions.Where(v => v != null);
-                    SelectedVersion =
-                        possibleVersions.FirstOrDefault(v => v.Version.Equals(_searchResultPackage.InstalledVersion))
-                        ?? possibleVersions.FirstOrDefault(v => v.IsValidVersion);
+                    SelectVersionNonPakageReferenceProject(latestVersion);
```

Comment:
Typo in the method name: 

`SelectVersionNonPakageReferenceProject` to `SelectVersionNonPackageReferenceProject`